mypy/test/testpep561.py

```diff
@@ -46,6 +46,18 @@ def virtualenv(python_executable: str = sys.executable) -> Iterator[tuple[str, s
             yield venv_dir, os.path.abspath(os.path.join(venv_dir, "bin", "python"))
 
 
+def upgrade_pip(python_executable: str, version_str: str) -> None:
```

Comment:
Let's skip this on Python 3.11 and newer, since this is really slow and venv is guaranteed to create an environment with new enough pip in 3.11

Reply:
Let's use this:
```
if (
    sys.version_info >= (3, 11)
    or (3, 10, 3) <= sys.version_info < (3, 11)
    or (3, 9, 11) <= sys.version_info < (3, 10)
    or (3, 8, 13) <= sys.version_info < (3, 9)
):
    return
```
I think worth it for how slow this is